src/app/[locale]/selected-filters.tsx

```diff
@@ -0,0 +1,114 @@
+import {SearchResultFilter} from '@/lib/dataset-fetcher';
+import {Dispatch} from 'react';
+import {useTranslations} from 'next-intl';
+import {XMarkIcon} from '@heroicons/react/24/solid';
+
+interface TagProps {
+  name?: string;
+  remove: () => void;
+}
+
+function Tag({name, remove}: TagProps) {
+  return (
+    <span className="flex items-center" data-test="selectedFilterTag">
+      <span className="m-1 inline-flex items-center rounded-full border border-gray-200 bg-white py-1.5 pl-3 pr-2 text-xs font-medium text-gray-900">
+        <span>{name}</span>
+        <button
+          data-test="removeSelectedFilter"
+          type="button"
+          className="ml-1 inline-flex flex-shrink-0 rounded-full p-1 text-gray-400 hover:bg-gray-200 hover:text-gray-500"
+          onClick={remove}
+        >
+          <XMarkIcon className="h-3 w-3" />
+        </button>
+      </span>
+    </span>
+  );
+}
+
+interface Props {
+  filters: {
+    searchResultFilters: SearchResultFilter[];
+    selectedFilters: string[];
+    setSelectedFilters: Dispatch<string[]>;
+  }[];
+  query: {
+    value: string;
+    setQuery: Dispatch<string>;
+  };
+}
+
+interface RemoveListItem {
+  id: string;
+  selectedFilters: string[];
+  setSelectedFilters: Dispatch<string[]>;
+}
+
+export default function SelectedFilters({filters, query}: Props) {
+  const t = useTranslations('Home');
+
+  function removeListItem({
+    id,
+    selectedFilters,
+    setSelectedFilters,
+  }: RemoveListItem) {
+    setSelectedFilters(selectedFilters.filter(filterId => id !== filterId));
+  }
+  function removeQuery() {
+    query.setQuery('');
+  }
+
+  function clearAllFilters() {
+    removeQuery();
+    filters.forEach(filter => {
+      filter.setSelectedFilters([]);
+    });
+  }
+
+  if (
+    !query.value &&
+    !filters.filter(filter => filter.selectedFilters.length).length
+  ) {
+    return null;
+  }
+  return (
+    <div className="mt-3 flex items-center">
+      <h3 className="text-xs font-medium text-gray-500">{t('filters')}</h3>
+      <div
+        aria-hidden="true"
+        className="hidden h-5 w-px bg-gray-300 sm:ml-3 sm:block mr-2"
+      />
+      <div className="flex flex-wrap grow">
+        {filters.map(
+          ({selectedFilters, searchResultFilters, setSelectedFilters}) =>
+            !!selectedFilters.length &&
+            selectedFilters.map(id => (
+              <Tag
+                key={id}
+                name={
+                  searchResultFilters.find(
+                    searchResultFilter => searchResultFilter.id === id
+                  )?.name
```

Comment:
To be sure: does `find()` always return a filter? Or could it return `undefined` (in which case `undefined.name` returns an error)?

Reply:
I should always return a filter. But in theory, `find` can always return undefined. So Typescript shows the error "Object is possibly 'undefined'.". Is there a way to do this differently?